packages/graphql/tests/integration/issues/1115.int.test.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { graphql, GraphQLSchema } from "graphql";
+import { Driver } from "neo4j-driver";
+import { Neo4jGraphQLAuthJWTPlugin } from "@neo4j/graphql-plugin-auth";
+import neo4j from "../neo4j";
+import { Neo4jGraphQL } from "../../../src";
+import { generateUniqueType } from "../../utils/graphql-types";
+import { runCypher } from "../../utils/run-cypher";
+import { createJwtRequest } from "../../utils/create-jwt-request";
+
+describe("https://github.com/neo4j/graphql/issues/1115", () => {
+    const parentType = generateUniqueType("Parent");
+    const childType = generateUniqueType("Child");
+
+    let schema: GraphQLSchema;
+    let driver: Driver;
+
+    beforeAll(async () => {
+        driver = await neo4j();
+
+        const typeDefs = `
+            type ${parentType} {
+                children: [${childType}!]! @relationship(type: "HAS", direction: IN)
+            }
+            type ${childType} {
+                tcId: String @unique
+            }
+
+            extend type ${childType}
+                @auth(
+                    rules: [
+                        { operations: [READ, CREATE, UPDATE, DELETE, CONNECT, DISCONNECT], roles: ["upstream"] }
+                        { operations: [READ], roles: ["downstream"] }
+                    ]
+                )
+        `;
+        const neoGraphql = new Neo4jGraphQL({
+            typeDefs,
+            driver,
+            plugins: {
+                auth: new Neo4jGraphQLAuthJWTPlugin({
+                    secret: "secret",
+                }),
+            },
+        });
+        schema = await neoGraphql.getSchema();
+    });
+
+    afterAll(async () => {
+        await driver.close();
```

Comment:
Consider cleaning up the created `parentType`s and `childType`s in the db.

Reply:
Do we want to do that cleanup here?

both parent and child are unique, and while it would make sense to clean the whole db at the end of the tests I'm worried that performing a cleanup on every test will lead to noticeable slower tests